va/va_test.go

```diff
@@ -450,19 +450,68 @@ func TestDCVAndCAASequencing(t *testing.T) {
 	test.AssertEquals(t, len(caaLog), 0)
 }
 
-func TestPerformRemoteValidationWithKnownOperation(t *testing.T) {
-	va, _ := setupWithRemotes(nil, "", []remoteConf{{ua: pass, rir: arin}, {ua: pass, rir: ripe}, {ua: pass, rir: apnic}}, nil)
-	op := func(ctx context.Context, rva RemoteVA, req proto.Message) (remoteResult, error) {
-		prob := &corepb.ProblemDetails{
-			ProblemType: string(probs.BadNonceProblem),
-			Detail:      "quite surprising",
-		}
-		return &vapb.ValidationResult{Problem: prob, Perspective: rva.Perspective}, nil
+func TestPerformRemoteOperation(t *testing.T) {
+	va, _ := setupWithRemotes(nil, "", []remoteConf{
+		{ua: pass, rir: arin},
+		{ua: pass, rir: ripe},
+		{ua: pass, rir: apnic},
```

Comment:
I like seeing the expanded test cases.

Looking again at the test I notice in the setup you provide only the RIRs, and not the perspectives. In the operation callbacks for each test case, you return only the perspective and not the RIR.

This works fine for now because we don't yet have the code that checks perspectives and RIRs. But I think pretty soon it would generate errors and need to be updated.

Probably the best thing here is to specify perspective in both the test remotes and the operation callback.

That means the operation callback has to have some way of knowing which perspective to return. Because this is a test, we can _cheat_: we can just take the perspective from the `rva RemoteVA` parameter and return that. But we will probably want to make that more sophisticated when we land the actual perspective-checking code.

Reply:
> Looking again at the test I notice in the setup you provide only the RIRs, and not the perspectives. In the operation callbacks for each test case, you return only the perspective and not the RIR.

Oops, good catch, I've gone ahead and fixed that.

> Probably the best thing here is to specify perspective in both the test remotes and the operation callback.

Perspective is set inside of setupRemotes():

https://github.com/letsencrypt/boulder/blob/8f250aecc283de637fd9c2fb73a20b638b343939/va/va_test.go#L197-L199

> That means the operation callback has to have some way of knowing which perspective to return. Because this is a test, we can cheat: we can just take the perspective from the rva RemoteVA parameter and return that. But we will probably want to make that more sophisticated when we land the actual perspective-checking code.

I'm a little unclear on this, lets chat offline so you can expand on that.

Reply:
Oops, no wonder it's unclear: I said "the operation callback has to have some way of knowing which perspective to return" but meant "the operation callback has to have some way of knowing which **RIR** to return".

And actually the answer was right under my nose: the operation callbacks were already pulling the expected Perspective from the RemoteVA object. I see in your latest revision they now also pull the RIR. 👍🏻